### PR TITLE
Rely on NodeJS API of Stylelint to execute CSS rules

### LINKED ISSUE #216: Analyze CSS and non-CSS files only part of SonarQube project
Currently we are analyzing all files in the project directory, we should analyze only those being part of the project (see https://jira.sonarsource.com/browse/MMF-1889)

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #216

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -38,9 +38,15 @@ only_sonarsource_qa: &ONLY_SONARSOURCE_QA
 
 build_task:
   gke_container:
-    <<: *CONTAINER_DEFINITION
+    dockerfile: .cirrus/nodejs-lts.Dockerfile
```
Comment (vilchik-elena) [inline .cirrus.yml]: due to TS analysis we require nodejs during analysis

File: its/plugin/src/test/java/org/sonar/css/its/IssuesTest.java [hunk 2]
```diff
@@ -56,12 +59,21 @@ public static void prepare() {
     orchestrator.getServer().associateProjectToQualityProfile(PROJECT_KEY, "css", "rules");
 
     SonarScanner scanner = Tests.createScanner(PROJECT_KEY);
+    scanner.setProperty("sonar.exclusions", "**/file-with-parsing-error-excluded.css");
     scanner.setProperty("sonar.html.file.suffixes", ".htm");
-    orchestrator.executeBuild(scanner);
+    scanner.setDebugLogs(true);
```
Comment (vilchik-elena) [inline its/plugin/src/test/java/org/sonar/css/its/IssuesTest.java]: @alban-auzeill We can remove this line

File: its/plugin/src/test/java/org/sonar/css/its/MetricsTest.java [hunk 3]
```diff
@@ -42,8 +42,8 @@ public static void prepare() {
 
   @Test
   public void test() {
-    assertThat(getProjectMeasureAsDouble("lines", PROJECT_KEY)).isEqualTo(32);
-    assertThat(getProjectMeasureAsDouble("ncloc", PROJECT_KEY)).isEqualTo(22);
+    assertThat(getProjectMeasureAsDouble("lines", PROJECT_KEY)).isEqualTo(43);
```
Comment (vilchik-elena) [inline its/plugin/src/test/java/org/sonar/css/its/MetricsTest.java]: increase due to the fact that PHP and HTML plugins are installed and they count LOCs for php and html files

File: its/plugin/src/test/java/org/sonar/css/its/NoCssFileProjectTest.java [hunk 4]
```diff
@@ -40,13 +41,13 @@
 
   @ClassRule
   public static Orchestrator orchestrator = Tests.ORCHESTRATOR;
+  public static BuildResult buildResult;
```
Comment (vilchik-elena) [inline its/plugin/src/test/java/org/sonar/css/its/NoCssFileProjectTest.java]: this property and it's assignment can be dropped

File: its/plugin/src/test/java/org/sonar/css/its/Tests.java [hunk 5]
```diff
@@ -55,6 +56,8 @@
 
   static {
     OrchestratorBuilder orchestratorBuilder = Orchestrator.builderEnv()
+      .addPlugin(MavenLocation.of("org.sonarsource.php", "sonar-php-plugin", "3.3.0.5166"))
```
Comment (saberduck) [inline its/plugin/src/test/java/org/sonar/css/its/Tests.java]: I am not sure, but perhaps we could use LATEST_RELEASE for other plugins, to make sure that we are always testing the current version

Comment (alban-auzeill) [reply its/plugin/src/test/java/org/sonar/css/its/Tests.java]: Good idea, I will try

File: sonar-css-plugin/css-bundle/src/server.ts [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        text: warning.text,
+        rule: warning.rule
+      })
+    );
+  });
+  return analysisResponse;
+}
+
+function getFileContent(filePath: string) {
+  const fileContent = fs.readFileSync(filePath, { encoding: "utf8" });
```
Comment (saberduck) [inline sonar-css-plugin/css-bundle/src/server.ts]: what about other encodings?

Comment (vilchik-elena) [reply sonar-css-plugin/css-bundle/src/server.ts]: I will do the same trick as in SonarJS (sending content only for different encoding or for sonarlint). But I decided to do it later to finally merge this PR

File: sonar-css-plugin/pom.xml [hunk 7]
```diff
@@ -147,25 +151,59 @@
             <plugin>
                 <groupId>com.github.eirslett</groupId>
                 <artifactId>frontend-maven-plugin</artifactId>
-                <version>1.4</version>
+                <version>1.6</version>
                 <configuration>
                     <workingDirectory>css-bundle</workingDirectory>
                     <installDirectory>target</installDirectory>
                     <nodeVersion>v8.9.4</nodeVersion>
```
Comment (saberduck) [inline sonar-css-plugin/pom.xml]: you can also upgrade Node version, this one is EOL soon

**Comment by alban-auzeill** [comment]
@vilchik-elena could you remove the 11 `.DS_Store`. I don't see why we need them?

**Comment by vilchik-elena** [comment]
LGTM

**Pull request merged**